engine_test.go

```diff
@@ -0,0 +1,38 @@
+package fuego
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestWithErrorHandler(t *testing.T) {
+	t.Run("default engine", func(t *testing.T) {
+		e := NewEngine()
+		err := NotFoundError{
+			Err: errors.New("Not Found :c"),
+		}
+		errResponse := e.ErrorHandler(err)
+		require.ErrorAs(t, errResponse, &HTTPError{})
+	})
+	t.Run("custom handler", func(t *testing.T) {
+		e := NewEngine(
+			WithErrorHandler(func(err error) error {
+				return errors.New("")
+			}),
+		)
```

Comment:
Please add a test about a custom error handler that could be used to invalidate an error by returning nil

Reply:
Why?

Reply:
Because people may do that in their custom handler if you dont want to support it, the mechanism of error handling should be rewritten to forbid this behavior

Reply:
That is not the responsibility of the function being tested. That would need to be tested [here](https://github.com/go-fuego/fuego/blob/2d6cfa42ba8daa774d1c34b7beea66c0772b478d/serve.go#L98) to give it any sort of meaning.

Reply:
You're totally right. But I'm not adding this functionality. I'm simply moving the function. I'd prefer an issue is opened to cover this case rather than it be within this PR as that is not the spirit of this unit of work.

Reply:
Yes, of course. I can understand that.

Let me open an issue